python/pyspark/testing/utils.py

```diff
@@ -209,3 +220,45 @@ def check_error(
         self.assertEqual(
             expected, actual, f"Expected message parameters was '{expected}', got '{actual}'"
         )
+
+
+def assertSparkSchemaEquality(
+    s1: Optional[Union[AtomicType, StructType, str, List[str], Tuple[str, ...]]],
+    s2: Optional[Union[AtomicType, StructType, str, List[str], Tuple[str, ...]]],
+):
+    if s1 != s2:
+        msg = "Schemas are different"
+        raise AssertionError(msg)
+
+
+def assertSparkDFEquality(
+    left: PySparkDataFrame, right: PySparkDataFrame,
+):
+    def assert_rows_equality(rows1, rows2):
+        if rows1 != rows2:
+            msg = "Dataframes are different"
+            raise AssertionError(msg)
+
+    left = left.sort(left.columns)
+    right = right.sort(right.columns)
+
+    assertSparkSchemaEquality(left.schema, right.schema)
```

Comment:
Column order is ignored here, which I don't think it's correct. Spark does not guarantee the order of results so you might need to collected results instead, but not the schema.